### ISSUE TITLE
[Feature Request]: Add the ability to create shortcuts for the games

### ISSUE THREAD

**Issue by 626164** [PCSX2/pcsx2#8349]
### Description

Would it be possible to add in the ability to create shortcuts for the games so we can have all our games on the Desktop and just double click to play? Instead of having to run PCSX2 first then launch a game.

### Reason

I'm making this request as I've made an emulator machine for a elder family member and it would make it infinitely easier on them to just double click the game icon to play it rather than launch this app then the game.

### Examples

An example from RPCS3, which has this feature
![Screenshot 2023-03-07 112916](https://user-images.githubusercontent.com/70711652/223486142-60ad93ff-c589-40e7-a8ff-828c936013e7.png)

**Comment by 626164** [comment]
That's a useful workaround to know about, thanks.

I'll leave the request open so we can hopefully get native support for shortcut creation.

**Comment by 626164** [comment]
> This can already be done in a similar fashion: https://wiki.pcsx2.net/Command-line_support
> 
> I know that's not what the request is about, but you have a workaround for it at the moment.

I'm trying to do this but keep getting "syntax is incorrect" errors. Where do you use the command link to create the shortcut? In windows command prompt? Or in the PCSX2 System Console? And can you provide an example of correctly working code to create a shortcut? 

Many thanks!

**Comment by refractionpcsx2** [comment]
you make a shortcut to the pcsx2 exe in windows, then you edit the shortcut and add the commandline options on to the end of the exe line

**Comment by 626164** [comment]
I added the commandline options as you said and when I try to run the shortcut I get a "Unknown Parameter: '-skipbios'" error

**Comment by refractionpcsx2** [comment]
Then don't try to use the example and use the actual list of commandline parameters.

**Comment by Jakey757** [comment]
-fullscreen -- "D:\PCSX2\games\Mark of kri, The (USA).iso"

don't include the brackets.

**Comment by 626164** [comment]
That did it! Thank you kindly!

If I were to have multiple parameters, would I just list them one after the other?

**Comment by Jakey757** [comment]
Yes, place them before the double dash -- anything before are options/parameters, anything after is supposed to be the game image path.

**Comment by refractionpcsx2** [comment]
none of the options are double hyphens anymore, please don't confuse them.

If you need further support, please use our discord, this isn't the place for it.

**Comment by Jakey757** [comment]
@refractionpcsx2 
" --: Signals that no more arguments will follow and the remaining parameters make up the filename."

im not saying to use it for the options, the wiki says to use it alone right before entering the path.

**Comment by refractionpcsx2** [comment]
I don't personally think there's any need, but ok.

**Comment by Jakey757** [comment]
hmm youre right, it works fine with or without it. not sure why the wiki includes it.

**Comment by RedDevilus** [comment]
Read carefully, the cli guide mentions double -- for old WX interface (exception -fullscreen) but Qt it's all single - for argument.

**Comment by CookiePLMonster** [comment]
> I don't personally think there's any need, but ok.

It's a niche feature but also without any "harm" to it when added - which makes it a fair request imo, even if just to provide a bit of extra feature parity across emus.

**Comment by Tyestor** [comment]
>I don't personally think there's any need, but ok.

If users cared about the opinions of programmers with regards to UX/UI everyone would work from linux terminals. But they don't because most people don't care about hyper optimised productivity they care about ease-of-use and convenience.

This suggestion is about ease of use and convenience.

**Comment by refractionpcsx2** [comment]
And if programmers listened to every whim of the user, the UX/UI would be an absolute mess, filled with a million niche settings making it confusing for everyone, so it goes both ways, so no need to be snarky.

But it's fine, I haven't written this feature off, else I would have closed it. If people find it generally useful then so be it.

**Comment by weirdbeardgame** [comment]
I think one use case would be for steam / added shortcuts in that regard

**Comment by RedDevilus** [comment]
I care about the UI/UX a ton, perhaps to pedantic levels, but there is no point just saying a remark that antagonizes not only the developers but also appreciative users. It makes sense even if niche and you could even make an easy batch script or integrate with other frontends as it's just a simple shortcut for them.

**Comment by NineKain** [comment]
I second this, is a way to make games more "real" in a sense in my opinion, gives them a lot more individuality

**Comment by stenzek** [comment]
As stated above, this can be done manually. IMO there's too many issues to bother:

1. Making it cross platform is an utter pain, unless Qt has some helper function to create it (I couldn't see any from a quick search). We _generally_ don't add features which are Windows-only.
2. As Silent said, it'd have the generic PCSX2 icon, not any game icon.
3. What options do you launch with? Some users will want windowed, others will want fullscreen, etc.
4. Do you launch the full Qt UI? Or hide the main game list (nogui mode)?
5. What about resuming a previous save-and-exit session?

With the last points, sure, you could have options for those, but our settings windows are already overcrowded as it is, let alone super niche stuff. The intended way to use PCSX2 is to open it, and select your game from the game list/grid. That way you can decide whether to resume where you left off, or start fresh, etc. We even offer a big picture UI so you can do that with controllers. I don't understand the fascination with having shortcuts/other launchers/etc.

**Comment by NineKain** [comment]
I am not anything remotely close to a dev/coder/IT guy so take my words with a pound of salt.

1. RPCS3 uses Qt, and its cross platform, and all its versions support this option, so maybe its easier than it seems
2. Cemu has this "issue" too, but its very easy to get icons on pages like steamgriddb.com, make them yourself or just leave it as default.
3. Default options + game by game options, it just skips opening the emu, and then opening the game (hence the "shortcut")
4. All 3 emus dont open the full UI, just the main window (which is basically one of the points of this feature imo) I think it should be left to dev discretion, but I'd prefer nogui.
5. That could be configurable, or just automatic, like I think you did for DuckStation, but could be worked out easily.

I understand it might seem a dumb feature for some, but having the acutal icons on the desktop makes them "feel" more "real" if that make sense, like you are playing a native game instead of emulating it, it is a minor thing by all means, but it offers a lot of customization to users, I attach an example of my own desktop.

![image](https://github.com/PCSX2/pcsx2/assets/72826232/1ac3126b-8811-40ac-ad0a-d6b5fefba49c)

**Comment by 1337ryan** [comment]
Is there any way to get this to boot a .py2 file on a pcsx2-qtx64-avx2 fork designed to do so? I keep getting "Failed to initialize CDVD".

**Comment by kamfretoz** [comment]
> Is there any way to get this to boot a .py2 file on a pcsx2-qtx64-avx2 fork designed to do so? I keep getting "Failed to initialize CDVD".

A quick [Google search](https://www.solvusoft.com/en/file-extensions/file-extension-py2/) reveals that a .py2 file is a python script file which has no bussiness running on PCSX2 (correct me on this one if its wrong as this is the only leads i could find.)

Also please use our Discord for support related question.

**Comment by 1337ryan** [comment]
> > Is there any way to get this to boot a .py2 file on a pcsx2-qtx64-avx2 fork designed to do so? I keep getting "Failed to initialize CDVD".
> 
> A quick [Google search](https://www.solvusoft.com/en/file-extensions/file-extension-py2/) reveals that a .py2 file is a python script file which has no bussiness running on PCSX2 (correct me on this one if its wrong as this is the only leads i could find.)
> 
> Also please use our Discord for support related question.

https://github.com/987123879113/pcsx2
"This is a fork of PCSX2 (the PS2 emulator) that works with Konami Python 1/2, Konami Bemani Python games, and the Namco System 246 games, which were based off Playstation 2 hardware."

I already have the game up and running perfectly. The .py2 file is recognized in the game list as a game and I can double click it to boot. I just can't get it to boot via this method of shortcut.

**Comment by F0bes** [comment]
We have nothing to do with that fork, we are upstream.
